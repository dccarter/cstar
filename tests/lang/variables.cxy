// global variables
var X = 10;
const Y = ("Hello", "World");

func main(args: [string]) {
    var x;
    var y = 10;
    const k = y;
    x = k;
    // variable scope
    {
        var k = 10;
        var y = x;
    }
    // typed
    {
        var x: i32;
        var y: (bool, i32) = (true, 20);
    }
}