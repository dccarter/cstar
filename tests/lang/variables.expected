const X = 10

var Y = ("Hello", "World")

func main(args: [string]) {
    const x
    const y = 10
    var k = y
    x = k
    {
        const k = 10
        const y = x
    }
}
